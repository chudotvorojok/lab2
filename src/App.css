.App {
    display:flex;
    justify-content:center;
    align-items:center;
    height: 100vh;
    width: 100%;
}
.calc-wrapper{
    width:300px;
    height:500px;
}
.row {
    display: flex;
    width: 100%;
}
  