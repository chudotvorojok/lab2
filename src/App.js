import React from 'react';
import calculate from './calculate';
import calculate_a from './calculate_a';
import calculate_b from './calculate_b';
import calculate_c from './calculate_c';
import "./App.css";


class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      firstnumber: '',
      secondnumber: '',
    };
    this.firstnumber_c = this.firstnumber_c.bind(this);
    this.secondnumber_c = this.secondnumber_c.bind(this);
    this.recalculate = this.recalculate.bind(this);
    this.recalculate_a = this.recalculate_a.bind(this);
    this.recalculate_b = this.recalculate_b.bind(this);
    this.recalculate_c = this.recalculate_c.bind(this);
  }

  recalculate() {
    this.setState(
      (state) => ({
        answerdivide: calculate(state.firstnumber, state.secondnumber),
      }),
    );
  }

  recalculate_a() {
    this.setState(
      (state) => ({
        answermultiply: calculate_a(state.firstnumber, state.secondnumber),
      }),
    );
  }

  recalculate_b() {
    this.setState(
      (state) => ({
        answeradd: calculate_b(state.firstnumber, state.secondnumber),
      }),
    );
  }

  recalculate_c() {
    this.setState(
      (state) => ({
        answersubtract: calculate_c(state.firstnumber, state.secondnumber),
      }),
    );
  }

  firstnumber_c(event) {
    this.setState({ firstnumber: event.target.value });
    this.recalculate();
    this.recalculate_a();
    this.recalculate_b();
    this.recalculate_c();
  }

  secondnumber_c(event) {
    this.setState({ secondnumber: event.target.value });
    this.recalculate();
    this.recalculate_a();
    this.recalculate_b();
    this.recalculate_c();
  }

  render() {
    return (
      <div className="input">
        <h4>
          Введите первое число:
          <input type="number" id="firstnumber" onInput={this.firstnumber_c} value={this.state.firstnumber} />
        </h4>
        <h4>
          Введите второе число:
          <input type="number" id="secondnumber" onChange={this.secondnumber_c} value={this.state.power} />
        </h4>
        <div id="answer" className="answerdivide">
          <h4>Частное:</h4>
          {this.state.answerdivide}
        </div>
        <div id="answer" className="answermultiply">
          <h4>Произведение:</h4>
          {this.state.answermultiply}
        </div>
        <div id="answer" className="answeradd">
          <h4>Сумма:</h4>
          {this.state.answeradd}
        </div>
        <div id="answer" className="answersubtract">
          <h4>Разность:</h4>
          {this.state.answersubtract}
        </div>
      </div>
    );
  }
}
export default App;
