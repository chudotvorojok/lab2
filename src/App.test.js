import React, { createElement } from 'react';
import renderer from 'react-test-renderer';
import App from './App';

describe('Рендеринг компонента', () => {
  test('Совпадение со снапшотом', () => {
    const app = renderer.create(<App />);
    expect(app.toJSON()).toMatchSnapshot();
  });
});
