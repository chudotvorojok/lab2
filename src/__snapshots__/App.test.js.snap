// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Проверка правильности расчётов 6+7 1`] = `
<div
  className="input"
>
  <h4>
    Введите первое число: 
    <input
      id="firstnumber"
      onInput={[Function]}
      type="number"
      value=""
    />
  </h4>
  <h4>
    Введите второе число: 
    <input
      id="secondnumber"
      onChange={[Function]}
      type="number"
    />
  </h4>
  <div
    className="answerdivide"
    id="answer"
  >
    <h4>
      Частное:
    </h4>
  </div>
  <div
    className="answermultiply"
    id="answer"
  >
    <h4>
      Произведение:
    </h4>
  </div>
  <div
    className="answeradd"
    id="answer"
  >
    <h4>
      Сумма:
    </h4>
  </div>
  <div
    className="answersubtract"
    id="answer"
  >
    <h4>
      Разность:
    </h4>
  </div>
</div>
`;

exports[`Проверка правильности расчётов Совпадение со снапшотом 1`] = `
<div
  className="input"
>
  <h4>
    Введите первое число: 
    <input
      id="firstnumber"
      onInput={[Function]}
      type="number"
      value=""
    />
  </h4>
  <h4>
    Введите второе число: 
    <input
      id="secondnumber"
      onChange={[Function]}
      type="number"
    />
  </h4>
  <div
    className="answerdivide"
    id="answer"
  >
    <h4>
      Частное:
    </h4>
  </div>
  <div
    className="answermultiply"
    id="answer"
  >
    <h4>
      Произведение:
    </h4>
  </div>
  <div
    className="answeradd"
    id="answer"
  >
    <h4>
      Сумма:
    </h4>
  </div>
  <div
    className="answersubtract"
    id="answer"
  >
    <h4>
      Разность:
    </h4>
  </div>
</div>
`;

exports[`Рендеринг компонента Совпадение со снапшотом 1`] = `
<div
  className="input"
>
  <h4>
    Введите первое число: 
    <input
      id="firstnumber"
      onInput={[Function]}
      type="number"
      value=""
    />
  </h4>
  <h4>
    Введите второе число: 
    <input
      id="secondnumber"
      onChange={[Function]}
      type="number"
    />
  </h4>
  <div
    className="answerdivide"
    id="answer"
  >
    <h4>
      Частное:
    </h4>
  </div>
  <div
    className="answermultiply"
    id="answer"
  >
    <h4>
      Произведение:
    </h4>
  </div>
  <div
    className="answeradd"
    id="answer"
  >
    <h4>
      Сумма:
    </h4>
  </div>
  <div
    className="answersubtract"
    id="answer"
  >
    <h4>
      Разность:
    </h4>
  </div>
</div>
`;
